processor.go

```diff
@@ -71,6 +71,7 @@ func NewProcessor(ctx gocontext.Context, hostname string, queue JobQueue,
 	buildJobsChan, err := queue.Jobs(ctx)
 	if err != nil {
 		context.LoggerFromContext(ctx).WithField("err", err).Error("couldn't create jobs channel")
+		cancel()
```

Comment:
I'm assuming this means it was missing and you noticed, but that it's not strictly related to the switch to stdlib `context`, ja? 😄